File: app/src/main/java/com/softhink/single/survey/PreferencesFragment.java
package com.softhink.single.survey;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageButton;

import com.appyvet.materialrangebar.RangeBar;
import com.softhink.single.BaseFragment;
import com.softhink.single.Constants;
import com.softhink.single.R;

/**
 * A simple {@link Fragment} subclass.
 */
public class PreferencesFragment extends BaseFragment
        implements View.OnClickListener,
        RangeBar.OnRangeBarChangeListener {

    private ImageButton btnNext;
    private RangeBar rangeBar;


    public PreferencesFragment() {
        // Required empty public constructor
    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_preferences, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        if (savedInstanceState == null) {
            rangeBar = view.findViewById(R.id.ageRangebar);
            btnNext = view.findViewById(R.id.btnNextPage);

            rangeBar.setRangePinsByValue(18F, 28F);
            rangeBar.setOnRangeBarChangeListener(this);
            btnNext.setOnClickListener(this);
        }

    }

    @Override
    public void onClick(View v) {
        switch (v.getId()){
            case R.id.btnNextPage:
                FragmentManager fragmentManager = getFragmentManager();
                fragmentManager.beginTransaction()
                        .setCustomAnimations(android.R.anim.slide_in_left, android.R.anim.slide_out_right)
                        .replace(R.id.containerPreferences, new InterestsFragment())
                        .addToBackStack(Constants.INSTANCE.getPREFERENCESFRAGMENT())
                        .commit();
                break;
        }
    }

    @Override
    public void onRangeChangeListener(RangeBar rangeBar, int leftPinIndex, int rightPinIndex, String leftPinValue, String rightPinValue) {

    }
}

File: app/src/main/java/com/softhink/single/survey/PreferencesFragment.kt
package com.softhink.single.survey

import android.os.Bundle
import androidx.annotation.NonNull
import androidx.annotation.Nullable
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageButton

import com.appyvet.materialrangebar.RangeBar
import com.softhink.single.BaseFragment
import com.softhink.single.Constants
import com.softhink.single.R
import kotlinx.android.synthetic.main.arrow_next.*
import kotlinx.android.synthetic.main.fragment_preferences.*


/**
 * A simple [Fragment] subclass.
 */
class PreferencesFragment : BaseFragment(), View.OnClickListener, RangeBar.OnRangeBarChangeListener {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_preferences, container, false)
    }

    override fun onViewCreated(@NonNull view: View, @Nullable savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        if (savedInstanceState == null) {
            ageRangebar.setRangePinsByValue(18f, 28f)
            ageRangebar.setOnRangeBarChangeListener(this)
            btnNextPage.setOnClickListener(this)
        }

    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.btnNextPage -> {
                val fragmentManager = fragmentManager
                fragmentManager!!.beginTransaction()
                        .setCustomAnimations(android.R.anim.slide_in_left, android.R.anim.slide_out_right)
                        .replace(R.id.containerPreferences, InterestsFragment())
                        .addToBackStack(Constants.PREFERENCESFRAGMENT)
                        .commit()
            }
        }
    }

    override fun onRangeChangeListener(rangeBar: RangeBar, leftPinIndex: Int, rightPinIndex: Int, leftPinValue: String, rightPinValue: String) {

    }
}// Required empty public constructor
